pcsx2/gui/GlobalCommands.cpp

```diff
@@ -150,15 +150,20 @@ namespace Implementations
 			OSDlog( Color_StrongRed, true, "(FrameLimiter) SlowMotion ENABLED." );
 			g_Conf->EmuOptions.GS.FrameLimitEnable = true;
 		}
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
+
 		pauser.AllowResume();
 	}
 
 	void Framelimiter_MasterToggle()
 	{
 		ScopedCoreThreadPause pauser;
 		g_Conf->EmuOptions.GS.FrameLimitEnable = !g_Conf->EmuOptions.GS.FrameLimitEnable;
-		GSsetVsync( g_Conf->EmuOptions.GS.FrameLimitEnable && g_Conf->EmuOptions.GS.VsyncEnable );
 		OSDlog( Color_StrongRed, true, "(FrameLimiter) %s.", g_Conf->EmuOptions.GS.FrameLimitEnable ? "ENABLED" : "DISABLED" );
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
```

Comment:
Anyone with a 120Hz/144Hz display that know how to code ?

It is possible to query the monitor mode with wx. `wxVideoMode wxDisplay::GetCurrentMode` or `wxArrayVideoModes wxDisplay::GetModes`.  Then `mode.GetRefresh()` could give us the refresh rate. It seems to be implemented only on windows. But I think it would make sense to not disable Vsync when turbo is enabled on high speed monitor.